tests/unit/test_stubs.py

```diff
@@ -21,3 +23,52 @@ def testing_connect(*args):
             vcr_connection.cassette = Cassette("test", record_mode=mode.ALL)
             vcr_connection.real_connection.connect()
             assert vcr_connection.real_connection.sock is not None
+
+    def test_body_consumed_once_stream(self, tmpdir, httpbin):
+        self._test_body_consumed_once(
+            tmpdir,
+            httpbin,
+            "body_consumed_once_stream.yml",
+            BytesIO(b"1234567890"),
+            BytesIO(b"9876543210"),
+            BytesIO(b"9876543210"),
+        )
+
+    def test_body_consumed_once_iterator(self, tmpdir, httpbin):
+        self._test_body_consumed_once(
+            tmpdir,
+            httpbin,
+            "body_consumed_once_iterator.yml",
+            iter([b"1234567890"]),
+            iter([b"9876543210"]),
+            iter([b"9876543210"]),
+        )
+
+    def _test_body_consumed_once(
+        self,
+        tmpdir,
+        httpbin,
+        testfile,
+        data1,
+        data2,
+        data3,
+    ):
+        testpath = str(tmpdir.join(testfile))
+        host, port = httpbin.host, httpbin.port
+        match_on = ["method", "uri", "body"]
+        with use_cassette(testpath, match_on=match_on):
+            conn1 = httplib.HTTPConnection(host, port)
+            conn1.request("POST", "/anything", body=data1)
+            resp1 = conn1.getresponse()
+            assert resp1.status == 501  # Chunked request Not implemented
+            conn2 = httplib.HTTPConnection(host, port)
+            conn2.request("POST", "/anything", body=data2)
+            resp2 = conn2.getresponse()
+            assert resp2.status == 501  # Chunked request Not implemented
+        with use_cassette(testpath, match_on=match_on) as cass:
+            conn3 = httplib.HTTPConnection(host, port)
+            conn3.request("POST", "/anything", body=data3)
+            resp3 = conn3.getresponse()
+            assert resp3.status == 501  # Chunked request Not implemented
+            assert cass.play_counts[0] == 0
+            assert cass.play_counts[1] == 1
```

Comment:
This secretly assumes and only works if `data2` and `data3` served the same data. Let's at least add a comment about that please if we cannot easily `assert` about due to `data2` and `data3` potentially being iterators.

Reply:
added a comment.